### ISSUE TITLE
Getting non-English character user name

### ISSUE THREAD

**Issue by yacaeh** [icza/screp#24]
Hi, I'm trying to get user name from the rep file and it returns "\ufffd\ufffdư" which seems default value for not getting character. 
How can I get correct encoded strings? 
Should I change https://github.com/icza/screp/blob/master/rep/header.go#L101 for this?

```func (h *Header) PlayerNames() string {
	buf := &strings.Builder{}
	var prevTeam byte
	for i, p := range h.Players {
		if i > 0 {
			if p.Team != prevTeam {
				buf.WriteString(" VS ")
			} else {
				buf.WriteString(", ")
			}
		}
		buf.WriteString(p.Name)
		prevTeam = p.Team
	}
	return buf.String()
}
```

**Comment by icza** [comment]
Strings stored in replays are interpreted as UTF-8 encoded sequences. When an invalid UTF-8 sequence is encountered, it is replaced with the Unicode replacement character `\ufffd`. This happens in the replay parser, so no matter what you do in `Header`, invalid sequences are already replaced there.

**Comment by yacaeh** [comment]
Thank you for the quick reply, so if I change this https://github.com/icza/screp/blob/master/repparser/repparser.go#L325 it would fix the problem?

**Comment by icza** [comment]
Yes, something like that. But what character encoding is used is not recorded in replays. Most of the time it's UTF-8. Do you have any information about this?

**Comment by yacaeh** [comment]
Hmm I guess it's also UTF-8, but it's output doesn't seem to match, `[185 246 198 176 0 188 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]`. There's no problem with Chat messages, but it's happening on names only. It's Korean BTW. Any ideas?

**Comment by icza** [comment]
This byte series is not valid UTF-8 data.

**Comment by icza** [comment]
After revisiting the `cString()` function, the data is not replaced in the repparser, it is just converted to `string`. But conversion does not replace invalid sequences, only when it is printed. So the data available in the `Header` is identical to the data available in the repparser (so you don't have to go down to the parser level).

**Comment by yacaeh** [comment]
Thanks, I've found it was encoded as euckr (cp949) :) it's converting successfully with that byte data set.

**Comment by icza** [comment]
Good to know. Did you decode bytes of player names in Header, or did you use the bytes in the replay parser level?

**Comment by yacaeh** [comment]
I've used bytes from `repparser`. In the header, it's already converted as '�'. It's a bit hacky way though, cause I cant' get encoding information. For my case I need only for Korean so I've done like this.

```

// cString returns a 0x00 byte terminated string from the given buffer.
func cString(data []byte) string {
	// Find 0x00 byte:
	checkUTF8, _ := DecodeRune(data)
	if checkUTF8 == 65533 {
		fmt.Print("is not UTF 8")
		return koreanString(data)
	}
	for i, ch := range data {
		if ch == 0 {
			return string(data[:i]) // excludes terminating 0x00
		}
	}
	// Couldn't find? As a fallback, just return the whole as-is:
	return string(data)
}

// return Korean String from given buffer
func koreanString(data []byte) string {
	euckrDec := korean.EUCKR.NewDecoder()
	euckrSrc := string(data)

	kString, _, err := transform.String(euckrDec, euckrSrc)
	res := strings.ReplaceAll(kString, "\u0000", "")
	res2 := strings.ReplaceAll(res, "�", "")
	if err != nil {
		panic(err)
	}
	return string(res2)
}

```
